src/domain/navigation/Navigation.ts

```diff
@@ -15,28 +15,53 @@ limitations under the License.
 */
 
 import {BaseObservableValue, ObservableValue} from "../../observable/ObservableValue";
+import type {allowsChild as AllowsChild} from "./index.js";
```

Comment:
I'd write the signature of the allowsChild in here as a type. Best not to have the definition be the implementation. You can probably also define it so it just takes keys of (a generic) SegmentType.